allennlp/modules/backbones/vilbert_backbone.py

```diff
@@ -157,16 +180,31 @@ def forward(
             extended_co_attention_mask,
         )
 
-        # Shape: (batch_size, num_tokens, embedding_dim)
+        # Shape: (batch_size * rolled_dimensions_product, num_tokens, embedding_dim)
         sequence_output_t = encoded_layers_t[:, :, :, -1]
-        # Shape: (batch_size, num_boxes, image_embedding_dim)
+        # Shape: (batch_size * rolled_dimensions_product, num_boxes, image_embedding_dim)
         sequence_output_v = encoded_layers_v[:, :, :, -1]
 
-        # Shape: (batch_size, pooled_output_dim)
+        # Shape: (batch_size * rolled_dimensions_product, pooled_output_dim)
         pooled_output_t = self.t_pooler(sequence_output_t)
-        # Shape: (batch_size, pooled_output_dim)
+        # Shape: (batch_size * rolled_dimensions_product, pooled_output_dim)
         pooled_output_v = self.v_pooler(sequence_output_v)
 
+        if rolled_dimensions:
+            unrolled_dimensions = [batch_size] + list(rolled_dimensions)
```

Comment:
Dimensions are usually tuples, but that's just convention.

Reply:
Oh I like tuples more, I'll switch it over